go/arrow/internal/arrdata/ioutil.go

```diff
@@ -191,3 +233,32 @@ func WriteStream(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.S
 		t.Fatal(err)
 	}
 }
+
+// WriteStreamCompressed writes a list of records to the given file descriptor as an ARROW stream
+// using the provided compression type.
+func WriteStreamCompressed(t *testing.T, f *os.File, mem memory.Allocator, schema *arrow.Schema, recs []array.Record, codec flatbuf.CompressionType) {
```

Comment:
same comment on the api.

Reply:
same as i mentioned above, this is not an externally facing Consumable API because it's an "internal" package which cannot be imported outside of this repo. The "internal" package is a special package that prevents anything below it from being an externally facing public API.